internal/engine/evaluator.go

```diff
@@ -662,14 +681,18 @@ func (er *PolicyEvalResult) unresolvedActions() []string {
 }
 
 // setEffect sets the effect for an action. DENY always takes precedence.
-func (er *PolicyEvalResult) setEffect(action string, effect EffectInfo) {
+func (er *PolicyEvalResult) setEffect(action string, scopeFallThrough policyv1.ScopeFallThrough, effect EffectInfo) {
 	delete(er.toResolve, action)
 
 	if effect.Effect == effectv1.Effect_EFFECT_DENY {
 		er.Effects[action] = effect
 		return
 	}
 
+	if effect.Effect == effectv1.Effect_EFFECT_ALLOW && scopeFallThrough == policyv1.ScopeFallThrough_SCOPE_FALL_THROUGH_ON_ALLOW {
+		effect.Effect = effectv1.Effect_EFFECT_NO_MATCH
```

Comment:
Wouldn't this ultimately lead to a DENY if none of the parents have a rule matching the action?

Reply:
Failed to write [this comment](https://github.com/cerbos/cerbos-private/issues/1244#issuecomment-2377302415) quick enough.

There's a whole subset of cases not covered at the moment which I'm tackling (sorry for premature draft PR)

Reply:
Just returning to this comment. Although I've since refactored this section out, wouldn't what you described have been the desired behaviour?

FALL_THROUGH_ON_ALLOW requires an explicit ALLOW from the ancestor policies in order to pass 🤔

Reply:
Actually, not in the case of the resource and principal policies in lieu of the removal of the key from `toResolve`, I think 🤔